### PR TITLE
RaR: Insight + typo fix in Yusuf/Musaazi

### PR DESCRIPTION
**Pull request by presheaf**
I'm happy to add a test if needed, just let me know. Also sorry about sneaking in a typo fix (missing space).

### CODE REVIEW

**Comment by NoahTheDuke** [comment]
Always love a good typo fix. This looks good! I'd much appreciate a test, but I understand how it is.

File: src/clj/game/cards/events.clj [hunk 1]
```diff
@@ -1051,6 +1051,19 @@
     :choices (req runnable-servers)
     :effect (effect (run target nil card))}
 
+   "Insight"
+   {:async true
+    :implementation "Rearranging R&D is not optional"
+    :effect (req
+             (let [from (take 4 (:deck corp))]
+               (if (pos? (count from))
```
Comment (Saintis) [inline src/clj/game/cards/events.clj]: This could be `when` if you don't have an `else` clause.

File: src/clj/game/cards/events.clj [hunk 2]
```diff
@@ -1051,6 +1051,19 @@
     :choices (req runnable-servers)
     :effect (effect (run target nil card))}
 
+   "Insight"
+   {:async true
+    :implementation "Rearranging R&D is not optional"
```
Comment (Saintis) [inline src/clj/game/cards/events.clj]: What does this mean? Is the Corp not allowed to re-arrange cards into the same arrangement?

File: src/clj/game/cards/events.clj [hunk 3]
```diff
@@ -1051,6 +1051,19 @@
...
+    :effect (req
+             (let [from (take 4 (:deck corp))]
+               (if (pos? (count from))
+                 (do (show-wait-prompt state :runner "Corp to rearrange the top cards of R&D")
```
Comment (Saintis) [inline src/clj/game/cards/events.clj]: Might want to say how many cards Corp is rearranging, so "top `(count from)` cards".

File: src/clj/game/cards/events.clj (hunk 2 above)
Comment (presheaf) [reply src/clj/game/cards/events.clj]: What I mean is that the card says the Corp 'may' rearrange, but I think asking the corp whether they want to will only cause confusion, so I assume the Corp wants to rearrange and rely on them to keep the order as-is if they don't want to. Maybe this note should just be deleted - seems like it causes confusion too. :)

**Pull request merged**